packages/core/collections/hash-dictionary.ts

```diff
@@ -100,3 +103,14 @@ export class HashDictionary<K extends OpaqueHash, V> implements ImmutableHashDic
     return this.map.delete(key.toString());
   }
 }
+
+import { BlobDictionary } from "./blob-dictionary.js";
+
+export class HashDictionary<K extends OpaqueHash, V>
+  extends BlobDictionary<K, V>
+  implements ImmutableHashDictionary<K, V>
+{
+  constructor() {
+    super(5);
```

Comment:
Would be good to move `5` to a constant and add some docs explaining the rationale.